### PR TITLE
Auth WIP: Database updates

### PR DESCRIPTION
**Pull request by gracekishino**
So far have just made updates to the database via the `schema.primsa`. 

@mdwiltfong just Wanted to check if you think this is OK before going any further?

### CODE REVIEW

File: server/prisma/schema.prisma [hunk 1]
```diff
@@ -36,10 +36,30 @@ model SessionUserGifts{
...
-  firstName                String
-  lastName                 String
+  firstName                String?
+  lastName                 String?
   password                 String
   sessionUserGifts         SessionUserGifts[]
+  tokens                   Token[]
+}
+
+model Token {
```
Comment (mdwiltfong) [inline server/prisma/schema.prisma:46]: Let's make this "Tokens" just to be consistent.

File: server/prisma/schema.prisma [hunk 2]
```diff
@@ -36,10 +36,30 @@ model SessionUserGifts{
...
+ id         Int       @id @default(autoincrement())
+ createdAt  DateTime  @default(now())
+ updatedAt  DateTime  @updatedAt
+ type       TokenType
```
Comment (mdwiltfong) [inline server/prisma/schema.prisma]: Can we switch the email token logic for a user being able to sign in with a username and password?

Comment (gracekishino) [reply server/prisma/schema.prisma]: @mdwiltfong thanks so much for all your help today! I updated the schema to use just one `jwtToken` and the user can login with the user and password.

**Comment by mdwiltfong** [comment]
Hey @gracekishino ! 

Sorry for hijacking this PR, but I think the backend on the main branch could really use it! Any chance I can go ahead and merge it? Let me know if there's something you want to throw in there.

Michael

**Comment by gracekishino** [comment]
@mdwiltfong go ahead and merge it! 

Hoping to update `server/prisma/index.ts` with the new DB changes at some point, but can create a new pull request for that 😊

**Comment by mdwiltfong** [comment]
> @mdwiltfong go ahead and merge it!
> 
> Hoping to update `server/prisma/index.ts` with the new DB changes at some point, but can create a new pull request for that 😊

Absolutely! Hopefully we'll have a chance to talk about it today during the JS Study session!

**Comment by gracekishino** [comment]
> > @mdwiltfong go ahead and merge it!
> 
> > 
> 
> > Hoping to update `server/prisma/index.ts` with the new DB changes at some point, but can create a new pull request for that 😊
> 
> 
> 
> Absolutely! Hopefully we'll have a chance to talk about it today during the JS Study session! 

Ok! Is there a session today? I couldn't see the event on the full stack page..

**Pull request merged**